src/g_editor.c

```diff
@@ -2397,7 +2397,25 @@ static void canvas_doclick(t_canvas *x, int xpos, int ypos, int which,
                 {
                     if (doit)
                     {
+                        //int issignal = obj_issignaloutlet(hitobj, closest);
                         int issignal = obj_issignaloutlet(hitobj, closest);
+                        int nchannels = obj_sigoutletgetchans(hitobj, closest);
```

Comment:
Actually, `issignal` is redundant because `obj_sigoutletgetchans` should return 0 for non-signal outlets. This means you can simply do the following:
```
int nchannels = obj_sigoutletgetchans(hitobj, closest);
int connection_thickness;
if (nchannels > 1)
    connection_thickness = 2.5 * x->gl_zoom;
else if (nchannels == 1)
    connection_thickness = 2 * x->gl_zoom;
else /* control outlet */
    connection_thickness = x->gl_zoom;
```
Or more tersely:
```
int nchannels = obj_sigoutletgetchans(hitobj, closest);
int connection_thickness =
    (nchannels > 1 ? 2.5 : nchannels == 1 ? 2.0 : 1.0) * x->gl_zoom;
```

Same in `tryconnect`